### PR TITLE
Added Compose Destinations library

### PR DESCRIPTION
**Pull request by sebvil**
Title. Feel free to review and if you don't like it/find another alternative, we can switch, but just opening the PR to show how the library looks like/how much easier it will be for us. I think the library does something very similar to our approach, but with KSP, so I'd rather use the third party library as it's less maintenance for us and code easier to write.

### CODE REVIEW

**Comment by sebvil** [comment]
Current dependencies on/for this PR:
* main
  * **PR #18** <a href="https://app.graphite.dev/github/pr/sebvil/mango/18" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sebvil/mango/18?utm_source=stack-comment).

File: app/build.gradle.kts [hunk 1]
```diff
@@ -86,10 +86,12 @@ dependencies {
...
     annotationProcessor(libs.room.compiler)
     ksp(libs.room.compiler)
-    implementation(project(":fakegen"))
-    ksp(project(":fakegen"))
     implementation(libs.kotlinx.coroutines.core)
+    implementation(libs.compose.destinations)
+    ksp(libs.compose.destinations.ksp)
 
+    implementation(project(":fakegen"))
+    kspDebug(project(":fakegen"))
```
Comment (sebvil) [inline app/build.gradle.kts:94]: We don't need fakes for prod builds, so only generate in debug builds

File: app/build.gradle.kts [hunk 2]
```diff
@@ -13,6 +13,7 @@ plugins {
 android {
     namespace = "com.sebastianvm.mango"
     compileSdk = 33
+    compileSdkPreview = "UpsideDownCakePrivacySandbox"
```
Comment (sebvil) [inline app/build.gradle.kts:16]: I was getting build errors without this.

Comment (amanzanero) [reply app/build.gradle.kts:16]: Ah, feel free to undo my changes! If you decide to, I would restack on main first

**Review by amanzanero — APPROVED** [review]
not opposed :) we can try it out

**Comment by amanzanero** [comment]
> so I'd rather use the third party library as it's less maintenance for us and code easier to write.

👍🏽 🔥

**Pull request merged**